> If I add env.NIX_CFLAGS_COMPILE = "-Wno-error=int-conversion"; to pkgs/top-level/perl-packages.nix line 11044, the Perl Packages build as well :)

It still fails with `goocanvasitemsimple.c:1483:15: error: assignment to 'GooCanvasItemModelSimple *' {aka 'struct _GooCanvasItemModelSimple *'} from incompatible pointer type 'GooCanvasItemModel *' {aka 'struct _GooCanvasItemModel *'}` for me

--- a/pkgs/development/libraries/goocanvas/default.nix
+++ b/pkgs/development/libraries/goocanvas/default.nix
@@ -9,13 +9,13 @@
   gnome,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "goocanvas";
   version = "1.0.0";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.bz2";
-    sha256 = "07kicpcacbqm3inp7zq32ldp95mxx4kfxpaazd0x5jk7hpw2w1qw";
+    url = "mirror://gnome/sources/goocanvas/${lib.versions.majorMinor finalAttrs.version}/goocanvas-${finalAttrs.version}.tar.bz2";
+    hash = "sha256-HAcu+IVnytJB+0rd7ibpvZZ0GxUD/3NtHBUvpthlcR4=";
   };
 
   nativeBuildInputs = [ pkg-config ];
@@ -25,9 +25,11 @@ stdenv.mkDerivation rec {
     glib
   ];
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=incompatible-pointer-types";
+
   passthru = {
     updateScript = gnome.updateScript {
-      packageName = pname;
+      packageName = "goocanvas";
       versionPolicy = "odd-unstable";
       freeze = true;
     };
@@ -39,4 +41,4 @@ stdenv.mkDerivation rec {
     license = licenses.lgpl2;
     platforms = lib.platforms.unix;
   };
-}
+})

--- a/pkgs/top-level/perl-packages.nix
+++ b/pkgs/top-level/perl-packages.nix
@@ -14917,6 +14917,7 @@ with self;
       pkgs.gtk2
       Gtk2
     ];
+    env.NIX_CFLAGS_COMPILE = "-Wno-error=int-conversion -Wno-error=implicit-function-declaration";
     meta = {
       description = "Perl interface to the GooCanvas";
       license = with lib.licenses; [

